File: app/src/main/java/nl/mranderson/pokefeeds/news/NewsContract.java
package nl.mranderson.pokefeeds.news;


import java.util.List;

interface NewsContract {

    interface View {

        void showLoadingState();

        void showExceptionState();

        void showEmptyState();

        void showListState(List<NewsItem> items);
    }

    interface Presenter {

        void attach(NewsContract.View view);

        void detach();

        void onEmptyButtonTapped();

        void onRetryButtonTapped();

        void onRefreshPulled();

        void onItemLinkTapped(String link);

    }

}

File: pokefeed/src/main/java/nl/mranderson/pokefeeds/news/NewsContract.kt
package nl.mranderson.pokefeeds.news

import io.reactivex.Single

internal interface NewsContract {

    interface Presenter {

        fun onEmptyButtonTapped()

        fun onRetryButtonTapped()

        fun onRefreshPulled()

        fun onItemLinkTapped(link: String)

        fun clear()
    }

    interface Interactor {
        fun getNews(url: String): Single<NewsResponse>
    }

}
